src/zenml/artifacts/load_directory_materializer.py

```diff
@@ -0,0 +1,98 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Only-load materializer for directories."""
+
+import os
+import tempfile
+from pathlib import Path
+from typing import Any, ClassVar, Tuple, Type
+
+from zenml.enums import ArtifactType
+from zenml.io import fileio
+from zenml.materializers.base_materializer import BaseMaterializer
+
+
+class PreexistingArtifactPath(type(Path())):  # type: ignore[misc]
```

Comment:
@bcdurak and me just looked at this again and thought that having an extra type just for this was still a little weird. What do you think of using `Path` as the `ASSOCIATED_TYPE`, and setting `SKIP_REGISTRATION=True` for this materializer?

This way, it would not be used when users return an artifact of type `Path`, but it will be used to load artifacts that were created by the `register_artifact` function because the materializer is specified explicitly.

Reply:
Yeah, this is great idea